test/e2e/prometheus_test.go

```diff
@@ -0,0 +1,312 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/common/expfmt"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const monitoringNamespace string = "monitoring"
+
+// Agent metrics to validate
+var antreaAgentMetrics = []string{
+	"antrea_agent_local_pod_count",
+	"antrea_agent_ovs_flow_table",
+	"antrea_agent_runtime_info",
+}
+
+// Controller metrics to validate
+var antreaControllerMetrics = []string{
+	"antrea_controller_address_group_processed",
+	"antrea_controller_address_group_sync_duration_milliseconds",
+	"antrea_controller_applied_to_group_processed",
+	"antrea_controller_applied_to_group_sync_duration_milliseconds",
+	"antrea_controller_length_address_group_queue",
+	"antrea_controller_length_applied_to_group_queue",
+	"antrea_controller_length_network_policy_queue",
+	"antrea_controller_network_policy_processed",
+	"antrea_controller_network_policy_sync_duration_milliseconds",
+	"antrea_controller_runtime_info",
+}
+
+var prometheusEnabled bool
+
+// Prometheus server JSON output
+type prometheusServerOutput struct {
+	Status string
+	Data   []map[string]string
+}
+
+func init() {
+	flag.BoolVar(&prometheusEnabled, "prometheus", false, "Enables Prometheus tests")
+}
+
+// skipIfPrometheusDisabled checks if Prometheus testing enabled, skip otherwise
+func skipIfPrometheusDisabled(t *testing.T) {
+	if !prometheusEnabled {
+		t.Skip("Prometheus testing is disabled")
+	}
+}
+
+// getMonitoringAuthToken retrieves monitoring authorization token, required for access to Antrea apiserver/metrics
+// resource
+func getMonitoringAuthToken(t *testing.T, data *TestData) string {
+	secrets, err := data.clientset.CoreV1().Secrets(monitoringNamespace).List(metav1.ListOptions{})
```

Comment:
can't we just get the secret by name now?

Reply:
Can I get it by name without knowing the name's generated suffix? Otherwise I should get the ServiceAccount to find the secret name.

Reply:
You're correct about the suffix.

I think my favorite solution would be to iterate over the secrets in the namespace and use the `kubernetes.io/service-account.name: prometheus` to identify the token (instead of relying on the prefix). I'm fine with postponing it to another PR.

Reply:
Done